demo/transient/transient_snes_mpfaof90.F90

```diff
@@ -376,6 +377,15 @@ program main
   call TDySetInitialCondition(tdy,U,ierr);
   CHKERRA(ierr);
 
+  ! Initialize the diagnostics DM and Vec, fetch the section that holds the
+  ! layout, and determine which diagnostic index corresponds to the liquid
+  ! mass (to check mass conservation).
+  call TDyCreateDiagnostics(tdy, diags_dm, ierr)
```

Comment:
Can `diags_dm` be part of the TDycore lib itself?

Reply:
It can. The reason I was doing things this way is to avoid incurring the cost of creating the diagnostics DM in cases where it's not necessary, and also to make clear that the DM can be manipulated by a user without affecting the dycore.

Reply:
> avoid incurring the cost of creating the diagnostics DM in cases where it's not necessary
Makes